### PR TITLE
Feature/28 Add USFM refs to sentences

### PR DESCRIPTION
**Pull request by ryderwishart**
This branch adds OSIS refs to the nodes trees and includes an XSL transformation that inserts these attributes in place for existing files. The XSL can be used in the future if we re-generate the nodes trees, in order to avoid having to modify other scripts used in the pipeline.

- [x] Amend this PR so that it generates USFM refs, including `!` word notation, instead of OSIS refs.

### CODE REVIEW

**Comment by klosoter** [comment]
This is probably easily solved by changing `"Node[m]"` to `"Node[m|c]"` on line 173. I still wonder _which_ id it would get.

**Comment by ryderwishart** [comment]
Thanks @klosoter, that is indeed an error. I was using an example file from Exodus that had no `<c>` nodes it turns out!

> This is probably easily solved by changing `"Node[m]"` to `"Node[m|c]"` on line 173. I still wonder _which_ id it would get.

Yes exactly what I was thinking. Since it would only match the `Node`, I would need to update this line in the XSL as well: 
```xml
<xsl:variable name="osisId" select="macula:osisId(data(./m/@n))"/>
```

**Comment by klosoter** [comment]
@ryderwishart I think that is the same solution I remember @jonathanrobie talking about concerning similar cases, but I'm not sure. I don't see anything against it, except that it probably needs to be documented clearly that this is a deviation.

**Comment by klosoter** [comment]
Oh, you mean to put the `@osisId` on ALL `m` elements, not just the ones inside a compound? Then, I don't think there should be any problem.

File: mappings/add-osis-refs-to-nodes.xsl [hunk 1]
```diff
@@ -170,9 +170,8 @@
         </xsl:copy>
     </xsl:template>    
     
-    <xsl:template match="Node[m]">
-        <!-- Add an osisId to every Node that has an m child -->
-        <xsl:variable name="osisId" select="macula:osisId(data(./m/@n))"/>
+    <xsl:template match="m">
+        <xsl:variable name="osisId" select="macula:osisId(data(./@n))"/>
```
[the remark this answers is not in the document]
Comment (klosoter) [reply mappings/add-osis-refs-to-nodes.xsl:174]: Looks all good now!

**Comment by ryderwishart** [comment]
As per @jonathanrobie's direction, instead of adding OSIS refs, I have instead added USFM refs to all of the Hebrew Node trees, both to the verses (`sentence`) and to the words (`m`).

**Pull request merged**